generator/src/main/java/com/scottlogic/deg/generator/decisiontree/ConstraintNode.java

```diff
@@ -35,5 +37,9 @@ static ConstraintNode merge(Iterator<ConstraintNode> constraintNodeIterator) {
 
         return new TreeConstraintNode(atomicConstraints, decisions);
     }
+
+    default Collection<Field> getConstrainedFields() {
```

Comment:
I'm not convinced this should be on a the `ConstraintNode` - can it be added as a static method in the `DecisionTreeCombinationProducer`?

This looks a bit like a helper method for the combination-producer, but is on the 'data' type.

If in future another type needs this functionality, we can move it back.

Reply:
Yeah it can be, I thought it would be useful in other places. I'll move it to the producer.